app/vmui/packages/vmui/src/hooks/useFetchQuery.ts

```diff
@@ -54,6 +57,7 @@ export const useFetchQuery = ({predefinedQuery, visible, display, customStep}: F
         const resp = await response.json();
         if (response.ok) {
           setError(undefined);
+          setTracingData(resp.trace);
```

Comment:
It would be better wrapping the original `trace` into a `Trace` object with `id`, `query` and `trace` fields as described in the comment above.

Reply:
used model